sound/soc/sof/intel/hda-stream.c

```diff
@@ -88,31 +88,46 @@ static int hda_setup_bdle(struct snd_sof_dev *sdev,
 /*
  * set up Buffer Descriptor List (BDL) for host memory transfer
  * BDL describes the location of the individual buffers and is little endian.
+ *
+ * The typical BDL entries after set up should looks like this:
+ *
+ * bdl[0] from the init_offset,  n entries before wrap,
+ * the e-1 full entries, and the last e entry.
+ *
+ *			     init_offset
+ *				 |
+ * |------|...........|---------S-------|------|........|--------|-------|
+ *  bdl[n] bdl[n+j]    bdl[e-1] | bdl[0] bdl[1] bdl[m]   bdl[n-2] bdl[n-1]
  */
 int hda_dsp_stream_setup_bdl(struct snd_sof_dev *sdev,
 			     struct snd_dma_buffer *dmab,
-			     struct hdac_stream *stream)
+			     struct hdac_stream *stream, u32 init_offset)
 {
 	struct sof_intel_dsp_bdl *bdl;
-	int i, offset, period_bytes, periods;
-	int remain, ioc;
+	unsigned int i, period_bytes, entries;
+	u32 bytes, offset, size = 0;
+	bool ioc, set_ioc, is_wrap = false;
 
```

Comment:
put on separate line if you initialize a variable.

Reply:
@plbossart I saw you mentioned this somewhere else also, can you share where do you get this recommendation, asking this as I am seeing actually many exceptions in the existed Kernel codes.

Reply:
it's not because others did things poorly in the past that they should be followed. The recommended practice is a single variable per line and x-mas tree shape.

Reply:
OK, let me change to follow this, good to know about this, thanks.